fluidkeys/secretsend.go

```diff
@@ -0,0 +1,108 @@
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/fluidkeys/fluidkeys/colour"
+
+	"github.com/fluidkeys/crypto/openpgp"
+	"github.com/fluidkeys/crypto/openpgp/armor"
+	"github.com/fluidkeys/fluidkeys/api"
+	"github.com/fluidkeys/fluidkeys/out"
+	"github.com/fluidkeys/fluidkeys/pgpkey"
+)
+
+func secretSend(recipientEmail string) exitCode {
+	out.Print("\n")
+	out.Print("[type or paste your message, ending by typing Ctrl-D]\n\n")
+
+	secret, err := scanUntilEOF()
+	if err != nil {
+		panic(err)
+		return 1
+	}
+
+	client := api.NewClient()
+	armoredPublicKey, _, err := client.GetPublicKey(recipientEmail)
+	if err != nil {
+		printFailed("Couldn't get the public key for " + recipientEmail + "\n")
+	}
+
+	printSuccess("Found public key for " + recipientEmail + "\n")
+
+	pgpKey, err := pgpkey.LoadFromArmoredPublicKey(armoredPublicKey)
+	if err != nil {
+		printFailed("Couldn't load the public key:")
+		out.Print("Error: " + err.Error() + "\n")
+		return 1
+	}
+
+	encryptedSecret, err := encryptSecret(secret, pgpKey)
+	if err != nil {
+		printFailed("Couldn't encrypt the secret:")
+		out.Print("Error: " + err.Error() + "\n")
+		return 1
+	}
+
+	response, err := client.CreateSecret(fmt.Sprintf("OPENPGP4FPR:%s", pgpKey.Fingerprint().Hex()), encryptedSecret)
+	if response.StatusCode != 201 || err != nil {
+		printFailed("Couldn't send the secret to " + recipientEmail)
+		if err != nil {
+			out.Print("Error: " + err.Error() + "\n")
+		}
+		return 1
+	}
+
+	printSuccess("Successfuly sent secret to " + recipientEmail + "\n")
+
+	out.Print("Tell them to get it by running\n")
+	out.Print("  " + colour.CommandLineCode("fk secret receive\n\n"))
+	return 0
+}
+
+func scanUntilEOF() (message string, err error) {
+	reader := bufio.NewReader(os.Stdin)
+	var output []rune
+
+	for {
+		input, _, err := reader.ReadRune()
+		if err != nil && err == io.EOF {
```

Comment:
this func can return an error, but as far as I can tell, there's no codepath which actually does that.

are you sure the `err != nil` and `err == io.EOF` should be handled identically?

Reply:
This is if there's an error *and* it's an EOF error. I could add a `panic` on an error that's not EOF?

Reply:
I think the caller already `panics` if you return an error, so returning an error seems more reasonable than panicking in here :)

Reply:
BTW you can safely test for `err == io.EOF` without `err != nil && `